### ISSUE TITLE
UnhandledPromiseRejectionWarning on an ubuntu ec2 instance

### ISSUE THREAD

**Issue by deepank411** [wellyshen/react-cool-starter#76]
Hi,

First of all, thank you for the excellent boilerplate. It has everything I need and more, much more!!

I need some assistance with an issue when I try to run the app on an ubuntu ec2 instance.

I ran it using `yarn start:production` and the error that was outputted is
```
(node:7985) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Exited with code 3
```

The base cause as highlighted in #31 is unhandled catch for Promise, but I couldn't find any unhandled Promises except in maybe `src/routes.js`.

The node version is `6.11.1`.
Any help would be great. Do please let me know if you need any other details.

**Comment by wellyshen** [comment]
Hi @deepank411, Thank you for give it a try, have you tried to handle the error catch of `./src/routes.js` yet?

**Comment by wellyshen** [comment]
@deepank411 , Have you wrote your own code or added any other library on it?

**Comment by deepank411** [comment]
No, I haven't. I cloned the project directly into the instance and tried running it unsuccessfully to the above error.

**Comment by wellyshen** [comment]
@deepank411 I will check that this weekend. Can I use AWS ec2 with free? I haven't experience of the service.

**Comment by deepank411** [comment]
A free tier is available to use for an year but requires a debit card for verification. [AWS Free Tier](https://aws.amazon.com/free/).

Thanks a lot! Will wait for a response.

**Comment by wellyshen** [comment]
@deepank411 I did some refine for the sever.js file, please check this changing could solve your problem or not. BTW, I didn't try on AWS ec2 instance (so many configurations I need to do for it, haha)

**Comment by deepank411** [comment]
@wellyshen No, it still throws the same error. Were you able to identify the place from where this error is getting generated. and why is it only coming on a server. It runs perfectly fine on my local system.

**Comment by wellyshen** [comment]
@deepank411 Actually, I have the same question with you. But for this issue, maybe you can use the [unhandledRejection](https://nodejs.org/api/process.html#process_event_unhandledrejection) event of node.js to handle the unhandled rejections.

**Comment by deepank411** [comment]
I figured out the problem, It was with the opn node module that automatically opens up the browser after building the project. That module cannot work on the server hence its giving an error. I commented `require('../tools/openBrowser').default(port);` and the error resolved, but I still can't access the application from a url. Do I need to do something more to deploy this application on a prod server?
